src/org/loklak/http/ClientConnection.java

```diff
@@ -272,7 +273,7 @@ public static String getRedirect(String urlstring, boolean useAuthentication) th
                 throw new IOException("redirect for  " + urlstring+ ": no location attribute found");
             } else {
                 EntityUtils.consumeQuietly(httpEntity);
-                throw new IOException("no redirect for  " + urlstring+ " fail: " + httpResponse.getStatusLine().getStatusCode() + ": " + httpResponse.getStatusLine().getReasonPhrase());
+                return urlstring;
```

Comment:
[This](https://github.com/loklak/loklak_server/blob/2abbf30e1a96f278ae06472d4f3ed98151c237ff/src/org/loklak/harvester/RedirectUnshortener.java#L76) is the function that calls `ClientConnection#getRedirect()` to get shortened URL. The `IOException` (which I removed) would be raised only if the status is non `30X`, which will be caught in caller function and the [latest rendered URL will be returned](https://github.com/loklak/loklak_server/blob/2abbf30e1a96f278ae06472d4f3ed98151c237ff/src/org/loklak/harvester/RedirectUnshortener.java#L89).

Now, if we replace the exception throwing by simply returning the `urlstring` (which is not semantically an issue as we tried to check redirect of a URL which doesn't redirect anywhere), the caller function will [break](https://github.com/loklak/loklak_server/blob/2abbf30e1a96f278ae06472d4f3ed98151c237ff/src/org/loklak/harvester/RedirectUnshortener.java#L82) and our job is done.

When the exception is thrown, the caller function panics due to it and [raises a false alarm in the logs](https://github.com/loklak/loklak_server/blob/2abbf30e1a96f278ae06472d4f3ed98151c237ff/src/org/loklak/harvester/RedirectUnshortener.java#L88). That's why I removed it.

But as I mentioned in #1114, there are cases when the response is non `30X` but redirect is made using `meta` tag. This can be tackled by extracting URL from the tag. But I think that it should go in another patch, hence haven't added it.